tests/ignite/distributed/utils/__init__.py

```diff
@@ -118,6 +118,15 @@ def _test_distrib_all_reduce(device):
         res = idist.all_reduce(t)
         assert res.device == t.device, f"{res.device} vs {t.device}"
 
+    # Test for all_reduce with group
+    if idist.get_world_size() > 2:
+        rank = idist.get_rank()
+        group = [1, 2]
+
+        t = torch.tensor([rank], device=idist.device())
+        res = idist.all_reduce(t, group=group)
+        assert res == torch.tensor([sum(group)])
```

Comment:
Have you checked whether this part of test is really executed ?
Maybe it would worth to add this into its own test.

Reply:
Yes, I checked locally. Let me separate to own test :)

Reply:
Let's create : 
1) `_test_distrib_all_reduce_group` test to check `idist.all_reduce` with list of int or real group (created with idist.new_group). Let's assume world_size=2
2) `_test_distrib_all_gather_group` test to check `idist.all_gather` with list of int or real group (created with idist.new_group). Let's assume world_size=2
3) `_test_distrib_new_group` test to check `idist.new_group`, you can inspire from https://github.com/pytorch/ignite/blob/b10b45a14720ced570a7ddd970d96d7c365caea4/tests/ignite/distributed/test_auto.py#L130-L149 for class type checking